### PR TITLE
(PUP-6122) Deprecate resource_types endpoint

### PR DESCRIPTION
**Pull request by fpringvaldsen**
Add a deprecation message to the resource_types endpoint.

### CODE REVIEW

File: lib/puppet/indirector/resource_type/parser.rb [hunk 1]
```diff
@@ -100,4 +100,9 @@ def resource_types_in(environment)
     environment.check_for_reparse
     environment.known_resource_types
   end
+
+  def allow_remote_requests?
+    Puppet.deprecation_warning("The resource_type endpoint is deprecated in favor of the environment_classes endpoint. See https://github.com/puppetlabs/puppet-server/blob/master/documentation/puppet-api/v3/environment_classes.md")
```
Comment (justinstoller) [inline lib/puppet/indirector/resource_type/parser.rb]: I don't know what our usual policy is on links within the code. Is there a docs.puppet.com link we could point to, or is pointing to the source code the best? If so, is it master we want to point to (won't that always be the documentation for an upcoming and unreleased version)?

Comment (kylog) [inline lib/puppet/indirector/resource_type/parser.rb]: A pretty common pattern is to register a links.puppetlabs.com link. Unfortunately that was broken in the move to puppet.com. Follow along (internal-only sorry) at https://tickets.puppetlabs.com/browse/OPS-8714.

**Comment by gguillotte** [comment]
Deprecated APIs have a docs page at https://docs.puppet.com/puppet/4.4/reference/deprecated_api.html. CC @jtappa

**Comment by KevinCorcoran** [comment]
The warning message looks good to me.  👍

**Comment by fpringvaldsen** [comment]
@kylog @justinstoller I've gone ahead and updated that message to contain a link to the latest docs on `docs.puppet.com` rather than to a docs page on the master branch of the puppetserver github repo. I would use a shortlink but since those are broken right now I figured this was best.

@gguillotte Do you need me to do anything with the `deprecated_api` docs page?

**Comment by kylog** [comment]
👍

**Comment by gguillotte** [comment]
@fpringvaldsen CC @jtappa for the docs. There at least needs to be release notes on the JIRA ticket.

**Comment by KevinCorcoran** [comment]
We look to have quorum for a merge.  @fpringvaldsen please be sure to add release notes to the ticket as @gguillotte requested.

**Pull request merged**